fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/TreeLike.java

```diff
@@ -328,19 +328,58 @@ default int height() {
     }
 
     /**
-     * returns the diameter of the tree. i.e. the maximum path between two leaves in the tree.
+     * Returns the diameter of the tree. i.e. the maximum path between two leaves in the tree.
+     *
      * @return the diameter of the tree.
      */
     default int diameter() {
+        return diameterWithConnectingEdgeFunction(current -> Math.min(Iterables.size(current.getChildren()), 2));
+    }
+
+    /**
+     * Calculates the diameter of the tree, incorporating the number of children of each node in the calculation.
```

Comment:
Consider to do this: Instead of adding `2` or the arity of the node, I think the proper way to do this would be:
1. calculate each (height/depth, diameter) for each child
2. for each child `i, j` pairing, compute the max of `(diameter_i, diameter_j, height_i + height_j + (j -i) + 2)` -- use the largest of these pairings.